### ISSUE TITLE
detector model remarks

### ISSUE THREAD

**Issue by egoosken** [flaport/photontorch#1]
I have 4 remarks on the current detector model.

1) sampling_rate/2 is not (necessarily) the highest frequency of the signal to be filtered.
In fact the sampling of the signal is done by the oscilloscope after the signal passed through the detector. It is best to therefor only use the concept of sample rate in the simulations after the detector. Until that moment one uses the concept of simulation timestep to denote the time between 2 simulation steps. This is however a pure simulation parameter and should not be linked to any physical property such as frequency.

I suggest removing the self.dt (sampling_timestep) and sampling_rate.

Add self.frequency and use this to calculate normal cutoff. In practice this self.frequency should be the bandwidth of the source as it is this that limits how fast the signal can vary in time.

2) If backpropagation through the filter is a necessity one can not simply use from scipy.signal import lfilter. This then requires writing the filter function in full with some adjustments inside the detector model which seriously hampers readability code (and risks introducing bugs). 

For the training phase of the reservoir however the detector model need not be included according to latest tests. This makes sense as the detector only adds white noise and filtering which can affect the final performance of the system but should not influence the training of the weights.  

Therefor I suggest reverting back to the use of the standard filter model directly imported from scipy.signal and disabling the detector during training.

NOTE: lfilter gives some error if signal is on cuda, thus I put signal momentarily on cpu and pass on self.device to detector model to put it filtered signal back on cuda afterwards. I suspect this problem can be avoided?

3) self.tau is not used anywhere in the code and can be removed.

4) If normal_cutoff >= 1 the filter produces an error both when using lfilter directly imported from scipy.signal or by using the back propagation compatible adjusted one. This was originally solved by putting normal_cutoff = 0.99. I recommend however to simply not apply the filter at all when normal_cutoff >=1 as error messages are usually generated for a reason and this seems like a bit of a hack. It should be investigated why exactly a normal_cutoff >=1 is not allowed as it seems that in such cases the filter should just let everything pass unaltered instead of trowing an error.

In summary I suggest adjusting the detector model as in the file attached.
[photodetector_model.zip](https://github.com/flaport/photontorch/files/3355129/photodetector_model.zip)
